### PR TITLE
Feature/14 rule use reducer

### PR DESCRIPTION
**Pull request by jasrial**

### CODE REVIEW

File: src/components/Rule.js [hunk 1]
```diff
@@ -1,66 +1,114 @@
...
+    newDimension: findDimensionsElement(dimensions, rule.tagType)
+      ? null
+      : rule.tagType,
+    targetDimension: findDimensionsElement(dimensions, rule.tagType)
+      ? rule.tagType
+      : null
+  };
+  const [state, dispatch] = useReducer(ruleReducer, initialState);
+  const {
+    regExpState,
```
Comment (vlabs) [inline src/components/Rule.js]: remove *State since you don't have a name clash now with props

File: src/components/Rule.js [hunk 2]
```diff
@@ -1,66 +1,114 @@
...
+  } = {
+    regExpState: state.rule.isRegexp,
+    matchState: state.rule.match,
+    tagTypeState: state.rule.tagType,
+    tagState: state.rule.tag,
+    newDimensionState: state.newDimension,
+    targetDimensionState: state.targetDimension
+  };
+
+  const handleFormChange = (e, label) => {
```
Comment (vlabs) [inline src/components/Rule.js]: can be shortened a lot
```
const handleFormChange = (event, label) => dispatch({ type: `set${label}Form`, value: event.currentTarget.value });
```

File: src/components/Rule.js [hunk 3]
```diff
@@ -1,66 +1,114 @@
...
+                handleSelectChange(selectedDimension)
+              }
+            />
+          </Col>
+          <Col md={4}>
+            <FormControl
+              placeholder="Or enter new dimension..."
+              disabled={targetDimensionState}
+              value={newDimensionState}
+              onChange={e => handleFormChange(e, "NewDimension")}
```
Comment (vlabs) [inline src/components/Rule.js]: don't use abbreviations: `e` should be `event`. please fix everywhere

File: src/components/Rule.js (hunk 1 above)
Comment (jasrial) [reply src/components/Rule.js]: cddd8e7b6c5cfbd38d72ef31bc0539f69d3fe41b

File: src/components/Rule.js (hunk 2 above)
Comment (jasrial) [reply src/components/Rule.js]: 81a9e72acdd62a5dada1b0b84bc0d5b912d47391

File: src/components/Rule.js (hunk 3 above)
Comment (jasrial) [reply src/components/Rule.js]: 3e8fc8131a77cb7720aee62bd153069725e2512a

File: src/components/Rule.js [hunk 4]
```diff
@@ -1,66 +1,102 @@
...
-        </Col>
-      </Row>
-    </Container>
-  </div>
-);
+const Rule = ({ rule, dimensions, onChange }) => {
+  const { tagType: tagTypeProp } = rule;
+  const initialState = {
+    rule,
+    newDimension: findDimensionElement(dimensions, tagTypeProp)
```
Comment (vlabs) [inline src/components/Rule.js]: rename to `findDimension`

Comment (jasrial) [reply src/components/Rule.js]: 983b5fcb7d9b6059b33e75841f509f59de96ebe5

**Pull request merged**